src/routes/boot.ts

```diff
@@ -780,14 +789,18 @@ const getFunnel = async (req: FastifyRequest, res: FastifyReply) => {
   // If the session id is set, we should use it to fetch the funnel data
   if (sessionId) {
     const sessionFunnel = await freyjaClient.getSession(sessionId);
-    if (sessionFunnel?.session?.userId === userId) {
+    if (
+      sessionFunnel?.session?.userId === userId &&
+      (!query?.id || sessionFunnel.funnel.id === query.id) &&
+      (!query.v || sessionFunnel.funnel.version === parseInt(query.v))
+    ) {
```

Comment:
this looks like hell, can restructure it a bit? For example split to multiple if statements that return undefined

Reply:
will do 🫡 
i'll add comments as well

Reply:
👌 awesome!